sendinblue.php

```diff
@@ -2832,6 +2957,59 @@ public function codeDeTracking()
+        if (!empty($camp->result->campaign_records)) {
+            foreach ($camp->result->campaign_records as $template_data) {
+                if ($template_data->templ_status === 'Active' && stristr($template_data->html_content, 'DOUBLEOPTIN') === false) {
+                    $options.= '<option value="' . $template_data->id . '"';
+                    if ($template_data->id == Configuration::get('Sendin_Template_Id', '', $this->id_shop_group, $this->id_shop)) {
+                        $options.= 'selected="selected"';
+                    }
+                    
+                    $options.= '>' . $template_data->campaign_name . '</option>';
+                }
+            }
+        }
+        $temp_data.= $options . '</select><span class="toolTip"
+        title="' . $this->l('Select a SendinBlue template that will be sent personalized for each contact that subscribes to your newsletter') . '"
+        ></span></div>';
+
+        $temp_confirm = '';
+        $temp_confirm.= '<div class="listData ' . $this->cl_version . ' managesubscribeBlock" style="text-align:left;"><select name="template_final" class="ui-state-default" style="width: 225px; height:22px; border-radius:4px; margin:10px 0;"><option value="">' . $this->l('Select Template') . '</option>
+        ';
+        $options = '';
+        $camp = $this->templateDisplay();
+        if (!empty($camp->result->campaign_records)) {
+            foreach ($camp->result->campaign_records as $template_data) {
+                if ($template_data->templ_status === 'Active' && stristr($template_data->html_content, 'DOUBLEOPTIN') === false) {
+                    $options.= '<option value="' . $template_data->id . '"';
+                    if ($template_data->id == Configuration::get('Sendin_Final_Template_Id', '', $this->id_shop_group, $this->id_shop)) {
+                        $options.= 'selected="selected"';
+                    }
+
+                    $options.= '>' . $template_data->campaign_name . '</option>';
+                }
+            }
+        }
+        $temp_confirm.= $options . '</select><span class="toolTip"
+        title="' . $this->l('Select a SendinBlue template that will be sent personalized for each contact that subscribes to your newsletter') . '"
+        ></span></div>';
+
+        $sendin_smtp_detail = Configuration::get('Sendin_Smtp_Result', '', $this->id_shop_group, $this->id_shop);
+        $smtp_data = Tools::jsondecode($sendin_smtp_detail);
+        $sendin_smtp = !empty($smtp_data->result->relay_data->status) ? $smtp_data->result->relay_data->status : '';
+        $smtp_alert = '';
+        if ($sendin_smtp !== 'enabled') {
+            $smtp_alert = '<div class="alert '. $this->cl_version .'"> ' . $this->l('You need an active SMTP (transactional) account to be able to send confirmation emails. Please').' <a href="mailto:contact@sendinblue.com">' . $this->l('contact customer service').'</a> ' . $this->l('to activate it.').'</div>';
+        }
+
```

Comment:
In order to follow the MVC architecture, HTML code should be generated in a tpl file, not in a PHP controller. You will see that your code in this file will be greatly lighter. :wink: